speech_recognition/__init__.py

```diff
@@ -1390,7 +1390,38 @@ def recognize_tensorflow(self, audio_data, tensor_graph='tensorflow-data/conv_ac
             for node_id in top_k:
                 human_string = self.tflabels[node_id]
                 return human_string
+            
+    def recognize_vosk(self, audio_data, language='en'):
+        from vosk import Model, KaldiRecognizer
 
+        if not os.path.exists("model"):
+            return "Please download the model from https://github.com/alphacep/vosk-api/blob/master/doc/models.md and unpack as 'model' in the current folder."
+            exit (1)
+
+        import pyaudio
+
+        model = Model("model")
+        rec = KaldiRecognizer(model, 16000)
+
+        p = pyaudio.PyAudio()
```

Comment:
Don't you need to use audio data here instead of microphone?

Reply:
This is a microphone test example from library: https://github.com/alphacep/vosk-api/blob/master/python/example/test_microphone.py

Reply:
Right, but audio data is already recorded in the main class and here you have just audio parameter of a method. You do not need to record once again.